### DISCUSSION TITLE
Choosing installed custom fonts at runtime

### DISCUSSION THREAD

**Discussion by Toafeek** [SixLabors/Fonts#169] [Q&A] [answered]
Users upload fonts from the front end and these fonts are stored in the cloud , when user wants to draw an image he chooses a custom font to be applied to text, and I get the custom font stream from the cloud and install it using the fontcollection install method,  The challenge I have is that when I call try find on a font I usually do not know the name of the font to find because during installation of fonts the family name is different some times, for example I install "RobotoBoldItalic" image sharp saves the family name as RobotoBold sets the available style to italic, at run time I do not know if I should search for robotobold or robotobolditalic .  please is there a way around this ?

**Comment by tocsoft** [comment, marked as answer]
When you call `FontCollection.Install(stream)`  you will have a `FontFamily` returned. Using the `FontFamily` you can new up a `Font` instance from it.

```c-sharp
var fonts = new FontCollection();
var fontFamily = fonts.Install(stream);
var pointSize = 36;
var font = new Font(fontFamily, pointSize);
```

As long as you only have a single font installed into the specific `FontCollection` you will get whatever variant you installed no matter the `FontStyle` specified. The way we currently use `FontStyle` is more of a hint than a specific requirement.

**Reply by Toafeek** [reply]
Thank you for the swift reply
In my situation I have robotobold and robotoitalic installed, the font collection contains one font family and two styles,
at run time the first style installed is applied, I want to be able to have an identifier for robotitalic and robotobold so I can choose the specific style I want

**Reply by tocsoft** [reply]
just don't install both fonts into the same collection?!... I'm confused why you are installing them both if you have a system where you are retrieving and installing a custom uploaded font. You obviously know exactly what font you want to use why would you install the 2nd variant? if you are wanting to use both for 2 separate drawing operations use a dedicated `FontCollection` for each.

**Reply by Toafeek** [reply]
Let me explain our situation and why we have it like that, we have multiple text we want to draw on the canvas, each text can have it's own font, so  we install all the fonts for each text before drawing them on the canvas, that is why we have two variant. From your last comment above I can see where our issue is coming from. if we install a font in the collection and use it immediately there will be no variant.  We wanted to avoid installing fonts every time we want to draw text on canvas, that is why we are installing all the fonts we need.

A work around which I implemented is to store each font installed in a dictionary with a key I know at runtime, so when I want to use the font I just have to call the dictionary key.

**Reply by Toafeek** [reply]
Thank you for the answer it was really helpful.

**Comment by Toafeek** [comment]
Hello, Please I will like to know if we can draw svg on canvas?
